gateway/ln-gateway/src/lib.rs

```diff
@@ -275,20 +280,75 @@ impl Gateway {
         }
     }
 
-    pub async fn run(self) -> anyhow::Result<TaskShutdownToken> {
+    pub async fn run(mut self) -> anyhow::Result<TaskShutdownToken> {
         let mut tg = TaskGroup::new();
 
-        run_webserver(self.password.clone(), self.listen, self.clone(), &mut tg)
-            .await
-            .expect("Failed to start webserver");
-        info!("Successfully started webserver");
-
+        self.start_webserver(&mut tg).await;
         self.start_gateway(&mut tg).await?;
         let handle = tg.make_handle();
         let shutdown_receiver = handle.make_shutdown_rx().await;
         Ok(shutdown_receiver)
     }
 
+    async fn start_webserver(&mut self, task_group: &mut TaskGroup) {
+        let gateway_db = self.gateway_db.clone();
+
+        let listen = self.listen;
+        let gateway = self.clone();
+        let cli_password = self.cli_password.clone();
+        let subgroup = task_group.make_subgroup().await;
+        task_group
+            .spawn("Webserver", move |handle| async move {
+                while !handle.is_shutting_down() {
+                    // Re-fetch the configuration because the password has changed.
+                    let gateway_config = Gateway::get_gateway_configuration(
+                        gateway_db.clone(),
+                        cli_password.clone(),
+                    )
+                    .await;
+                    let mut webserver_group = subgroup.make_subgroup().await;
+                    run_webserver(
+                        gateway_config.clone(),
+                        listen,
+                        gateway.clone(),
+                        &mut webserver_group,
+                    )
+                    .await
+                    .expect("Failed to start webserver");
+                    info!("Successfully started webserver");
+
+                    gateway_db
+                        .wait_key_check(&GatewayConfigurationKey, |v| {
+                            v.filter(|cfg| {
+                                if let Some(old_config) = gateway_config.clone() {
+                                    old_config.password != cfg.clone().password
+                                } else {
+                                    true
+                                }
+                            })
+                        })
+                        .await;
+
+                    info!("GatewayConfiguration has been updated, restarting webserver...");
```

Comment:
That you have to do that is kidna weird, but ok, let's iterate.

Reply:
Yeah, couldn't figure out a way to overwrite the bearer token without restarting the webserver. If we find a better way let's definitely change it!